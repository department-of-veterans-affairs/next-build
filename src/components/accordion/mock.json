{
  "type": "paragraph--basic_accordion",
  "id": "ca7c4618-6701-4f84-a636-8209c05c9eca",
  "drupal_internal__id": 38313,
  "drupal_internal__revision_id": 284296,
  "langcode": "en",
  "status": true,
  "created": "2021-04-28T01:16:36+00:00",
  "parent_id": "16142",
  "parent_type": "node",
  "parent_field_name": "field_content_block",
  "behavior_settings": [],
  "default_langcode": true,
  "revision_translation_affected": true,
  "content_translation_source": "und",
  "content_translation_outdated": false,
  "content_translation_changed": null,
  "field_header": "What are Vet Centers?",
  "field_rich_wysiwyg": {
    "value": "<p>Vet Centers are small, non-medical, counseling centers conveniently located in your community. They're staffed by highly trained counselors and team members dedicated to seeing you through the challenges that come with managing life during and after the military.</p>\r\n\r\n<p>Whether you come in for one-on-one counseling or to participate in a group session, at Vet Centers you can form social connections, try new things, and build a support system with people who understand you and want to help you succeed.</p>\r\n",
    "format": "rich_text_limited",
    "processed": "<p>Vet Centers are small, non-medical, counseling centers conveniently located in your community. They're staffed by highly trained counselors and team members dedicated to seeing you through the challenges that come with managing life during and after the military.</p>\n<p>Whether you come in for one-on-one counseling or to participate in a group session, at Vet Centers you can form social connections, try new things, and build a support system with people who understand you and want to help you succeed.</p>\n"
  },
  "links": {
    "self": {
      "href": "https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/basic_accordion/ca7c4618-6701-4f84-a636-8209c05c9eca?resourceVersion=id%3A284296"
    }
  },
  "paragraph_type": {
    "type": "paragraphs_type--paragraphs_type",
    "id": "5f3ba120-da59-4c9a-b7d8-5fd72bf3eecd",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": "basic_accordion"
    }
  },
  "relationshipNames": ["paragraph_type"]
}
