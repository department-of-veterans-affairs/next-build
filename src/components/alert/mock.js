export const mockResponse = {
  type: 'paragraph--alert',
  id: '019070f3-9e3e-4c69-83e4-c7eace8c414f',
  drupal_internal__id: 6003,
  drupal_internal__revision_id: 27807,
  langcode: 'en',
  status: true,
  created: '2019-09-04T13:45:29+00:00',
  parent_id: '876',
  parent_type: 'node',
  parent_field_name: 'field_content_block',
  behavior_settings: [],
  default_langcode: true,
  revision_translation_affected: true,
  content_translation_source: 'und',
  content_translation_outdated: false,
  content_translation_changed: null,
  field_alert_heading: 'Blue Water Navy Veterans and Agent Orange',
  field_alert_type: 'information',
  links: {
    self: {
      href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/alert/019070f3-9e3e-4c69-83e4-c7eace8c414f?resourceVersion=id%3A27807',
    },
  },
  paragraph_type: {
    type: 'paragraphs_type--paragraphs_type',
    id: 'daa6bb41-1b12-46f7-a355-96ceb6fa18dc',
    resourceIdObjMeta: {
      drupal_internal__target_id: 'alert',
    },
  },
  field_alert_block_reference: null,
  field_va_paragraphs: [
    {
      type: 'paragraph--wysiwyg',
      id: 'abb45dcb-b33f-4080-b20f-2f89d4281806',
      drupal_internal__id: 6004,
      drupal_internal__revision_id: 27808,
      langcode: 'en',
      status: true,
      created: '2019-09-04T13:45:29+00:00',
      parent_id: '6003',
      parent_type: 'paragraph',
      parent_field_name: 'field_va_paragraphs',
      behavior_settings: [],
      default_langcode: true,
      revision_translation_affected: true,
      content_translation_source: 'und',
      content_translation_outdated: false,
      content_translation_changed: null,
      field_wysiwyg: {
        value:
          '<div class="usa-alert-body" ><p>Blue Water Navy Veterans are now entitled to a presumption of service connection for illnesses related to Agent Orange exposure. This is a result of <a href="https://www.congress.gov/bill/116th-congress/house-bill/299/text" target="_blank" rel="noopener ">Public Law 116-23</a>, the Blue Water Navy Vietnam Veterans Act of 2019. The law was signed on June 25, 2019, and takes effect on January 1, 2020.</p><p>Read below to learn more about eligibility requirements and how to file a claim. If we denied your claim in the past, you can file a new claim based on this change in law.</p></div>',
        format: 'rich_text',
        processed:
          '<p>Blue Water Navy Veterans are now entitled to a presumption of service connection for illnesses related to Agent Orange exposure. This is a result of <a href="https://www.congress.gov/bill/116th-congress/house-bill/299/text" target="_blank">Public Law 116-23</a>, the Blue Water Navy Vietnam Veterans Act of 2019. The law was signed on June 25, 2019, and takes effect on January 1, 2020.</p><p>Read below to learn more about eligibility requirements and how to file a claim. If we denied your claim in the past, you can file a new claim based on this change in law.</p>',
      },
      links: {
        self: {
          href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/wysiwyg/abb45dcb-b33f-4080-b20f-2f89d4281806?resourceVersion=id%3A27808',
        },
      },
      resourceIdObjMeta: {
        target_revision_id: 27808,
        drupal_internal__target_id: 6004,
      },
      paragraph_type: {
        type: 'paragraphs_type--paragraphs_type',
        id: '885e4b61-cfd2-44b9-94ae-f068ba2b48b6',
        resourceIdObjMeta: {
          drupal_internal__target_id: 'wysiwyg',
        },
      },
      relationshipNames: ['paragraph_type'],
    },
  ],
  relationshipNames: [
    'paragraph_type',
    'field_alert_block_reference',
    'field_va_paragraphs',
  ],
}
