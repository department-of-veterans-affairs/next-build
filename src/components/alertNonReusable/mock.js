export const mockResponse = {
  type: 'paragraph--non_reusable_alert',
  id: '4eff76c1-5e3e-410b-a56a-64b572f98756',
  drupal_internal__id: 13072,
  drupal_internal__revision_id: 145559,
  langcode: 'en',
  status: true,
  created: '2020-10-01T17:08:59+00:00',
  parent_id: '13073',
  parent_type: 'paragraph',
  parent_field_name: 'field_alert_non_reusable_ref',
  behavior_settings: [],
  default_langcode: true,
  revision_translation_affected: null,
  content_translation_source: 'und',
  content_translation_outdated: false,
  content_translation_changed: null,
  field_alert_heading: 'Sample non-reusable alert',
  field_alert_type: 'information',
  links: {
    self: {
      href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/non_reusable_alert/4eff76c1-5e3e-410b-a56a-64b572f98756?resourceVersion=id%3A145559',
    },
  },
  paragraph_type: {
    type: 'paragraphs_type--paragraphs_type',
    id: '3d633fc0-97d6-4087-b709-edfa2d1e6a1e',
    resourceIdObjMeta: {
      drupal_internal__target_id: 'non_reusable_alert',
    },
  },
  field_va_paragraphs: [
    {
      type: 'paragraph--wysiwyg',
      id: '2fc092f1-969a-4136-9039-4fc50a9f93c4',
      drupal_internal__id: 13071,
      drupal_internal__revision_id: 141493,
      langcode: 'en',
      status: true,
      created: '2020-10-01T17:09:17+00:00',
      parent_id: '13072',
      parent_type: 'paragraph',
      parent_field_name: 'field_va_paragraphs',
      behavior_settings: [],
      default_langcode: true,
      revision_translation_affected: true,
      content_translation_source: 'und',
      content_translation_outdated: false,
      content_translation_changed: null,
      field_wysiwyg: {
        value: '<p>This is where the text goes for the alert</p>\r\n',
        format: 'rich_text',
        processed: '<p>This is where the text goes for the alert</p>',
      },
      links: {
        self: {
          href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/wysiwyg/2fc092f1-969a-4136-9039-4fc50a9f93c4?resourceVersion=id%3A141493',
        },
      },
      resourceIdObjMeta: {
        target_revision_id: 141493,
        drupal_internal__target_id: 13071,
      },
      paragraph_type: {
        type: 'paragraphs_type--paragraphs_type',
        id: '885e4b61-cfd2-44b9-94ae-f068ba2b48b6',
        resourceIdObjMeta: {
          drupal_internal__target_id: 'wysiwyg',
        },
      },
      relationshipNames: ['paragraph_type'],
    },
  ],
  relationshipNames: ['paragraph_type', 'field_va_paragraphs'],
}
