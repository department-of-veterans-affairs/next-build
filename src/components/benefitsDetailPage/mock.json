{
  "type": "node--page",
  "id": "27aaaef6-0c67-4639-b7d7-84cb3c124cff",
  "drupal_internal__nid": 298,
  "drupal_internal__vid": 304204,
  "status": true,
  "title": "Check Your VA Claim or Appeal Status",
  "created": "2019-04-24T19:05:07+00:00",
  "changed": "2020-04-30T20:11:16+00:00",
  "moderation_state": "published",
  "path": {
    "alias": "/check-your-va-claim-or-appeal-status",
    "pid": 479,
    "langcode": "en"
  },
  "breadcrumbs": [
    {
      "uri": "http://va-gov-cms.ddev.site/",
      "title": "Home",
      "options": []
    },
    {
      "uri": "http://va-gov-cms.ddev.site/check-your-va-claim-or-appeal-status",
      "title": "Check Your VA Claim or Appeal Status",
      "options": []
    }
  ],
  "field_description": "Find out how to check the status of a VA claim or appeal online.",
  "field_intro_text_limited_html": {
    "value": "Find out how to check the status of a VA claim or appeal online.",
    "format": "rich_text_limited",
    "processed": "<p>Find out how to check the status of a VA claim or appeal online.</p>\n"
  },
  "field_table_of_contents_boolean": true,
  "field_administration": {
    "type": "taxonomy_term--administration",
    "id": "2c331a6d-b525-4f0c-8bea-4ecde41c7ef0",
    "drupal_internal__tid": 1,
    "name": "Veterans Benefits Administration",
    "description": null,
    "path": {
      "alias": "/vba"
    }
  },
  "field_alert": null,
  "field_content_block": [
    {
      "type": "paragraph--wysiwyg",
      "id": "b871d67e-44a1-45d7-a3c9-948795c0b10d",
      "drupal_internal__id": 3885,
      "field_wysiwyg": {
        "value": "<h2>How to check your VA claim or appeal status online</h2><p>Sign in to VA.gov to check the status of your VA claim or appeal.</p>",
        "format": "rich_text",
        "processed": "<h2>How to check your VA claim or appeal status online</h2><p>Sign in to VA.gov to check the status of your VA claim or appeal.</p>"
      }
    }
  ],
  "field_featured_content": [
    {
      "type": "paragraph--wysiwyg",
      "id": "07b2ad1b-9b94-464a-8961-3ca73e84c8aa",
      "drupal_internal__id": 3886,
      "field_wysiwyg": {
        "value": "<h3>What you need to know</h3><p>You can check the status of your disability claim online.</p>",
        "format": "rich_text",
        "processed": "<h3>What you need to know</h3><p>You can check the status of your disability claim online.</p>"
      }
    }
  ],
  "field_related_links": null
}
