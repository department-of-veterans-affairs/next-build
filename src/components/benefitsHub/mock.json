{
  "type": "node--landing_page",
  "id": "fd822f25-788f-489c-a0fc-b1f7ad6c20d5",
  "path": {
    "alias": "/health-care",
    "pid": 123,
    "langcode": "en"
  },
  "breadcrumbs": [
    {
      "uri": "https://www.va.gov/",
      "title": "Home",
      "options": []
    }
  ],
  "drupal_internal__nid": 67,
  "drupal_internal__vid": 1234,
  "langcode": "en",
  "status": true,
  "title": "VA health care",
  "created": "2020-01-01T00:00:00+00:00",
  "changed": "2023-12-01T12:00:00+00:00",
  "promote": false,
  "sticky": false,
  "default_langcode": true,
  "revision_translation_affected": true,
  "moderation_state": "published",
  "metatag": [],
  "field_related_office": null,
  "field_alert": null,
  "field_title_icon": "",
  "field_home_page_hub_label": "Health care",
  "field_teaser_text": "Apply for and manage VA health care benefits",
  "field_links": [],
  "field_description": "Get VA health care coverage for preventive care, primary care, specialty care, and more.",
  "field_meta_title": "VA Health Care | Veterans Affairs",
  "field_intro_text": "With a VA health care plan, you're covered for regular checkups with your primary care provider and appointments with specialists (like cardiologists, gynecologists, and mental health providers).",
  "field_plainlanguage_date": "",
  "field_promo": null,
  "field_related_links": null,
  "field_spokes": null,
  "field_support_services": []
}
