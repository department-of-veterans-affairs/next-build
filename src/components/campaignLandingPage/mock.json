{
  "type": "node--campaign_landing_page",
  "id": "7f375d16-00c5-4aed-a1ff-c01e4c4254c3",
  "drupal_internal__nid": 48974,
  "drupal_internal__vid": 679789,
  "langcode": "en",
  "revision_timestamp": "2022-09-01T23:13:15+00:00",
  "revision_log": null,
  "status": false,
  "title": "PACT Act and survivors",
  "created": "2022-08-30T17:28:02+00:00",
  "changed": "2022-09-01T23:13:15+00:00",
  "promote": false,
  "sticky": false,
  "default_langcode": true,
  "revision_translation_affected": true,
  "breadcrumbs": [
    {
      "uri": "https://va-gov-cms.ddev.site/",
      "title": "Home",
      "options": []
    },
    {
      "uri": "https://va-gov-cms.ddev.site/initiatives/pact-act-and-survivors",
      "title": "PACT Act and Survivors",
      "options": []
    }
  ],
  "moderation_state": "draft",
  "expiration_date": null,
  "warning_date": null,
  "metatag": [
    {
      "tag": "meta",
      "attributes": {
        "name": "title",
        "content": "PACT Act and survivors | Veterans Affairs"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "name": "description",
        "content": "Edit introduction"
      }
    },
    {
      "tag": "link",
      "attributes": {
        "rel": "image_src",
        "href": "https://www.va.gov/img/design/logo/va-og-image.png"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "property": "og:site_name",
        "content": "Veterans Affairs"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "property": "og:title",
        "content": "PACT Act and survivors | Veterans Affairs"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "property": "og:image",
        "content": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/3_2_medium_thumbnail/public/2021-03/Listening-Session-Website-Graphic-730x370.jpg"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "property": "og:image:alt",
        "content": "U.S. Department of Veterans Affairs"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "name": "twitter:card",
        "content": "summary_large_image"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "name": "twitter:site",
        "content": "@DeptVetAffairs"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "name": "twitter:title",
        "content": "PACT Act and survivors | Veterans Affairs"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "name": "twitter:image",
        "content": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/3_2_medium_thumbnail/public/2021-03/Listening-Session-Website-Graphic-730x370.jpg"
      }
    },
    {
      "tag": "meta",
      "attributes": {
        "name": "twitter:image:alt",
        "content": "U.S. Department of Veterans Affairs"
      }
    }
  ],
  "path": {
    "alias": "/initiatives/pact-act-and-survivors",
    "pid": 66745,
    "langcode": "en"
  },
  "content_translation_source": "und",
  "content_translation_outdated": false,
  "field_clp_events_header": null,
  "field_clp_events_panel": false,
  "field_clp_faq_panel": true,
  "field_clp_resources_header": null,
  "field_clp_resources_intro_text": null,
  "field_clp_resources_panel": false,
  "field_clp_spotlight_header": null,
  "field_clp_spotlight_intro_text": null,
  "field_clp_spotlight_panel": false,
  "field_clp_stories_cta": null,
  "field_clp_stories_header": null,
  "field_clp_stories_intro": null,
  "field_clp_stories_panel": false,
  "field_clp_video_panel": false,
  "field_clp_video_panel_header": null,
  "field_clp_what_you_can_do_header": "Edit Heading",
  "field_clp_what_you_can_do_intro": "Edit Introduction",
  "field_clp_why_this_matters": "Edit introduction",
  "field_hero_blurb": "This page will help answer your questions about the PACT Act and what it means for survivors.",
  "field_last_saved_by_an_editor": null,
  "links": {
    "self": {
      "href": "https://va-gov-cms.ddev.site/jsonapi/node/campaign_landing_page/7f375d16-00c5-4aed-a1ff-c01e4c4254c3?resourceVersion=id%3A679789"
    }
  },
  "node_type": {
    "type": "node_type--node_type",
    "id": "ffad462a-e060-445c-bdc1-4456ce404059",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": "campaign_landing_page"
    }
  },
  "revision_uid": {
    "type": "user--user",
    "id": "5ef9f654-32dc-4c55-91c5-74a170edc16e",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": 4028
    }
  },
  "uid": {
    "type": "user--user",
    "id": "daf963c4-14fc-4a02-8925-c030df3a3276",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": 1451
    }
  },
  "field_administration": {
    "type": "taxonomy_term--administration",
    "id": "dfdffc79-de5d-4c1d-a3eb-fa8865f9e72a",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": 1011
    }
  },
  "field_benefit_categories": [
    {
      "type": "node--landing_page",
      "id": "ef71edfa-9fe8-4d31-b84c-270af94539c4",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 809
      }
    }
  ],
  "field_clp_audience": [
    {
      "type": "taxonomy_term--audience_beneficiaries",
      "id": "386eb70d-696c-4af3-8986-306ce63d90de",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 268
      }
    }
  ],
  "field_clp_events_references": [],
  "field_clp_faq_cta": null,
  "field_clp_faq_paragraphs": [
    {
      "type": "paragraph--q_a",
      "id": "eb782434-d0c6-4df8-9bf0-908bf70b7508",
      "resourceIdObjMeta": {
        "target_revision_id": 727810,
        "drupal_internal__target_id": 114124
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "bab5963c-fe40-4c90-a785-7724c6272f65",
      "resourceIdObjMeta": {
        "target_revision_id": 727812,
        "drupal_internal__target_id": 114126
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "75d9a2b3-a193-441f-b3e9-b06e06cff59b",
      "resourceIdObjMeta": {
        "target_revision_id": 727814,
        "drupal_internal__target_id": 114128
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "707b8f59-2e59-45e9-82e2-75148d901f88",
      "resourceIdObjMeta": {
        "target_revision_id": 727816,
        "drupal_internal__target_id": 114130
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "e7314214-96f1-406b-aead-241c3a68cd3c",
      "resourceIdObjMeta": {
        "target_revision_id": 727818,
        "drupal_internal__target_id": 114132
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "cd49e354-81fe-4dbd-b2f0-cd03b7210fb6",
      "resourceIdObjMeta": {
        "target_revision_id": 727820,
        "drupal_internal__target_id": 114134
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "109816a4-8f6d-4035-bd04-6dcc8b246b4d",
      "resourceIdObjMeta": {
        "target_revision_id": 727822,
        "drupal_internal__target_id": 114136
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "95378d3c-77cd-4acf-aa57-34d5d0bd149a",
      "resourceIdObjMeta": {
        "target_revision_id": 727824,
        "drupal_internal__target_id": 114138
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "23a24ae2-99c3-4b7a-915c-d37a2cf3de2f",
      "resourceIdObjMeta": {
        "target_revision_id": 727826,
        "drupal_internal__target_id": 114140
      }
    },
    {
      "type": "paragraph--q_a",
      "id": "d75ad221-0b9b-4e1d-b3f8-c6a305bfde25",
      "resourceIdObjMeta": {
        "target_revision_id": 727828,
        "drupal_internal__target_id": 114142
      }
    }
  ],
  "field_clp_resources": [],
  "field_clp_resources_cta": null,
  "field_clp_reusable_q_a": null,
  "field_clp_spotlight_cta": null,
  "field_clp_spotlight_link_teasers": [],
  "field_clp_stories_teasers": [],
  "field_clp_video_panel_more_video": null,
  "field_clp_what_you_can_do_promos": [
    {
      "type": "block_content--promo",
      "id": "56cf3784-3186-4bb9-8866-582d59b14f02",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 65
      }
    }
  ],
  "field_connect_with_us": null,
  "field_hero_image": {
    "type": "media--image",
    "id": "f2cd9371-cff5-418b-9e15-ad631e8c4686",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": 2689
    }
  },
  "field_media": null,
  "field_primary_call_to_action": {
    "type": "paragraph--button",
    "id": "8b849a21-d0af-49ed-92e4-ff19598d5851",
    "resourceIdObjMeta": {
      "target_revision_id": 727829,
      "drupal_internal__target_id": 113967
    }
  },
  "field_secondary_call_to_action": null,
  "relationshipNames": [
    "node_type",
    "revision_uid",
    "uid",
    "field_administration",
    "field_benefit_categories",
    "field_clp_audience",
    "field_clp_events_references",
    "field_clp_faq_cta",
    "field_clp_faq_paragraphs",
    "field_clp_resources",
    "field_clp_resources_cta",
    "field_clp_reusable_q_a",
    "field_clp_spotlight_cta",
    "field_clp_spotlight_link_teasers",
    "field_clp_stories_teasers",
    "field_clp_video_panel_more_video",
    "field_clp_what_you_can_do_promos",
    "field_connect_with_us",
    "field_hero_image",
    "field_media",
    "field_primary_call_to_action",
    "field_secondary_call_to_action"
  ]
}
