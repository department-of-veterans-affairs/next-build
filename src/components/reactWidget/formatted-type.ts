import { PublishedParagraph } from '@/types/formatted/publishedEntity'

export type ReactWidget = PublishedParagraph & {
  type: 'paragraph--react_widget'
  entityId: number | null
  widgetType: string
  ctaWidget?: boolean
  loadingMessage?: string
  slowLoadingMessage?: string
  timeout?: number
  errorMessage?: string
  defaultLink?: {
    url: string
    title: string
  } | null
  buttonFormat?: boolean
}
