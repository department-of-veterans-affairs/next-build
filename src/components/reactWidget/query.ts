import { ParagraphReactWidget } from '@/types/drupal/paragraph'
import { QueryFormatter } from 'next-drupal-query'
import { ReactWidget } from '@/components/reactWidget/formatted-type'

export const formatter: QueryFormatter<ParagraphReactWidget, ReactWidget> = (
  entity: ParagraphReactWidget
) => {
  return {
    type: entity.type as ReactWidget['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    widgetType: entity.field_widget_type,
    ctaWidget: entity.field_cta_widget,
    loadingMessage: entity.field_loading_message,
    timeout: entity.field_timeout,
    errorMessage: entity.field_error_message.processed,
    defaultLink: entity.field_default_link
      ? {
          url: entity.field_default_link.url,
          title: entity.field_default_link.title,
        }
      : null,
    buttonFormat: entity.field_button_format,
  }
}
