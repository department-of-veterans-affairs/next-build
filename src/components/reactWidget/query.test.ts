/**
 * @jest-environment node
 */

import { ParagraphReactWidget } from '@/types/drupal/paragraph'
import { queries } from '@/data/queries'
import { mockResponse } from '@/mocks/reactWidget.mock'

const ReactWidgetMock: ParagraphReactWidget = mockResponse

describe('React widget formatData', () => {
  test('outputs formatted data', () => {
    expect(
      queries.formatData('paragraph--react_widget', ReactWidgetMock)
    ).toMatchSnapshot()
  })

  test('handles null defaultLink correctly', () => {
    const modifiedMock = { ...ReactWidgetMock, field_default_link: null }
    const formattedData = queries.formatData(
      'paragraph--react_widget',
      modifiedMock
    )
    expect(formattedData.defaultLink).toBeNull()
  })
})
