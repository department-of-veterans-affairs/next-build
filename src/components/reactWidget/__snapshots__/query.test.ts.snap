// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`React widget formatData outputs formatted data 1`] = `
{
  "buttonFormat": true,
  "ctaWidget": false,
  "defaultLink": {
    "title": "Apply for Veterans Pension Benefits",
    "url": "/pension/application/527EZ",
  },
  "entityId": 4102,
  "errorMessage": "<strong>Weâ€™re sorry. Something went wrong when we tried to load your saved application.</strong><br>Please try refreshing your browser in a few minutes.",
  "id": "5663a0dd-a69c-40b7-b50b-526646e32204",
  "loadingMessage": "Checking your application status.",
  "timeout": 20,
  "type": "paragraph--react_widget",
  "widgetType": "health-care-app-status",
}
`;
